From 85e43bd88028238e3c92eeeba084375e42a75fbf Mon Sep 17 00:00:00 2001
From: SeungYoon Lee <w22905@motorola.com>
Date: Fri, 16 Sep 2016 10:59:20 -0500
Subject: [PATCH 938/959] IKCTODSP-75 ASoC: msm8952: ospl: add sonification
 table

add one more tuning table loading mechanism, sonification,
to fully meet the SOP requirement.

Change-Id: If4efdb1f044c31d8eed171697ed27c75d219a073
Signed-off-by: SeungYoon Lee <w22905@motorola.com>
Reviewed-on: https://gerrit.mot.com/900317
SLTApproved: Slta Waiver <sltawvr@motorola.com>
SME-Granted: SME Approvals Granted
Tested-by: Jira Key <jirakey@motorola.com>
Reviewed-by: Igor Kovalenko <igork@motorola.com>
Submit-Approved: Jira Key <jirakey@motorola.com>
---
 sound/soc/msm/ospl2xx.c | 15 +++++++++++----
 1 file changed, 11 insertions(+), 4 deletions(-)

diff --git a/sound/soc/msm/ospl2xx.c b/sound/soc/msm/ospl2xx.c
index 0a9463d1478..56a8d73cecf 100644
--- a/sound/soc/msm/ospl2xx.c
+++ b/sound/soc/msm/ospl2xx.c
@@ -24,9 +24,11 @@ static char const *ospl2xx_ext_config_tables[] = {
 	"opalum.rx.ext.config.0",
 	"opalum.rx.ext.config.1",
 	"opalum.rx.ext.config.2",
+	"opalum.rx.ext.config.3",
 	"opalum.tx.ext.config.0",
 	"opalum.tx.ext.config.1",
 	"opalum.tx.ext.config.2",
+	"opalum.tx.ext.config.3",
 };
 
 static int ext_config_loaded;
@@ -448,7 +450,7 @@ int ospl2xx_afe_get_param(uint32_t param_id)
 	return result;
 }
 
-#define NUM_RX_CONFIGS 3
+#define NUM_RX_CONFIGS 4
 static int ext_rxconfig;
 static int ext_txconfig = NUM_RX_CONFIGS;
 static int int_rxconfig;
@@ -671,9 +673,10 @@ static char const *ospl2xx_rx_int_config_index[] = {
 	"opalum.rx.int.config.0",
 	"opalum.rx.int.config.1",
 	"opalum.rx.int.config.2",
+	"opalum.rx.int.config.3",
 };
 static const struct soc_enum ospl2xx_rx_int_config_enum[] = {
-	SOC_ENUM_SINGLE_EXT(3, ospl2xx_rx_int_config_index),
+	SOC_ENUM_SINGLE_EXT(NUM_RX_CONFIGS, ospl2xx_rx_int_config_index),
 };
 
 /* PARAM_ID_OPALUM_RX_RUN_CALIBRATION */
@@ -730,6 +733,7 @@ static char const *ospl2xx_rx_ext_config_text[] = {
 	"opalum.rx.ext.config.0",
 	"opalum.rx.ext.config.1",
 	"opalum.rx.ext.config.2",
+	"opalum.rx.ext.config.3",
 };
 static const struct soc_enum ospl2xx_rx_ext_config_enum[] = {
 	SOC_ENUM_SINGLE_EXT(NUM_RX_CONFIGS,
@@ -780,6 +784,7 @@ static char const *ospl2xx_audio_mode_text[] = {
 	"NORMAL",
 	"VOICE",
 	"RING",
+	"SONIFICATION",
 };
 static const struct soc_enum ospl2xx_audio_mode_enum[] = {
 	SOC_ENUM_SINGLE_EXT(NUM_RX_CONFIGS,
@@ -942,7 +947,7 @@ static int ospl2xx_tx_get_temp_cal(struct snd_kcontrol *kcontrol,
 }
 
 /* PARAM_ID_OPALUM_TX_SET_USE_CASE */
-#define NUM_TX_CONFIGS 3
+#define NUM_TX_CONFIGS 4
 static int ospl2xx_tx_int_config_get(struct snd_kcontrol *kcontrol,
 			struct snd_ctl_elem_value *ucontrol)
 {
@@ -976,9 +981,10 @@ static char const *ospl2xx_tx_int_config_index[] = {
 	"opalum.tx.int.config.0",
 	"opalum.tx.int.config.1",
 	"opalum.tx.int.config.2",
+	"opalum.tx.int.config.3",
 };
 static const struct soc_enum ospl2xx_tx_int_config_enum[] = {
-	SOC_ENUM_SINGLE_EXT(3, ospl2xx_tx_int_config_index),
+	SOC_ENUM_SINGLE_EXT(NUM_TX_CONFIGS, ospl2xx_tx_int_config_index),
 };
 
 /* PARAM_ID_OPALUM_TX_SET_EXTERNAL_CONFIG */
@@ -1011,6 +1017,7 @@ static char const *ospl2xx_tx_ext_config_text[] = {
 	"opalum.tx.ext.config.0",
 	"opalum.tx.ext.config.1",
 	"opalum.tx.ext.config.2",
+	"opalum.tx.ext.config.3",
 };
 static const struct soc_enum ospl2xx_tx_ext_config_enum[] = {
 	SOC_ENUM_SINGLE_EXT(NUM_TX_CONFIGS,
-- 
2.11.0

